src/main/java/com/google/sps/utility/DirectionsUtility.java

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.utility;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.client.utils.URIBuilder;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/** Utility class for the Google Directions API. */
+public final class DirectionsUtility {
+
+  /** Prevents instances of this class. */
+  private DirectionsUtility() {}
+
+  /**
+   * Sets multiple waypoints to the waypoints parameter of the Google Directions API.
+   *
+   * @param waypoints A list of string containing the multiple waypoints to set as parameter values.
+   * @return A string of waypoints to set to the waypoints parameter.
+   */
+  private static String buildWaypoints(List<String> waypoints) {
+    StringBuilder waypointsBuilder = new StringBuilder("optimize:true");
+    if (waypoints.isEmpty()) {
+      return waypointsBuilder.toString();
+    }
+    for (String waypoint : waypoints) {
+      waypointsBuilder.append("|" + waypoint);
+    }
+    return waypointsBuilder.toString();
+  }
+
+  /**
+   * Builds custom URI to send a HTTP GET Request to the Google Directions API.
+   *
+   * @param destination A string representing the destination to get directions to.
+   * @param origin A string representing the origin to get directions from.
+   * @param waypoints A list of string consisting of waypoints to visit between the destination and
+   *     the origin.
+   * @param apiKey A string representing the API key to authenticate a Google Directions API call.
+   * @return A string representing the URI.
+   * @throws URISyntaxException Indicates that the built string could not be parsed as a URI.
+   */
+  public static String getDirectionsUri(
+      String destination, String origin, List<String> waypoints, String apiKey)
```

Comment:
Does this method need to be accessible outside of the class?

Reply:
Nope, the only reason this is public was only for testing purposes. Should probably test these within this class. Thanks for pointing it out!

Reply:
If there could be a use for this outside of the class, you could consider making this public.